src/Host.cpp

```diff
@@ -2598,3 +2628,956 @@ void Host::setCompactInputLine(const bool state)
         }
     }
 }
+
+QPointer<TConsole> Host::findConsole(QString name)
+{
+    if (name.isEmpty() or name == QStringLiteral("main")) {
+        return &*mpConsole;
+    } else {
+        return mpConsole->mSubConsoleMap.value(name);
+    }
+}
+
+QPair<bool, QStringList> Host::getLines(const QString& windowName, const int lineFrom, const int lineTo)
+{
+    if (!mpConsole) {
+        QStringList failMessage;
+        failMessage << QStringLiteral("internal error: no main TConsole - please report").arg(windowName);
+        return qMakePair(false, failMessage);
+    }
+
+    if (windowName.isEmpty() || windowName == QLatin1String("main")) {
+        return qMakePair(true, mpConsole->getLines(lineFrom, lineTo));
+    }
+
+    auto pC = mpConsole->mSubConsoleMap.value(windowName);
+    if (pC) {
+        return qMakePair(true, pC->getLines(lineFrom, lineTo));
+    } else {
```

Comment:
I have no problems with else-after-return. I only have a problem with else-after-return where one is an error and one is not; and in that case the `if` should check for the error so that you can follow the logic more easily by simply skipping it. Some editors have fold-in-the-block commands for that which are really useful if the code is large enough to warrant them.

I (or somebody else) can certainly revisit the code and fix all of these later.